ui/components/component-library/avatar-icon/avatar-icon.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { Ref } from 'react';
+import classnames from 'classnames';
+import {
+  BorderColor,
+  DISPLAY,
+  AlignItems,
+  JustifyContent,
+  BackgroundColor,
+  IconColor,
+  TextColor,
+} from '../../../helpers/constants/design-system';
+
+import { Icon, IconSize } from '../icon';
+import { AvatarBase } from '../avatar-base';
+
+import { AvatarIconProps, AvatarIconSize } from './avatar-icon.types';
+
+export const AvatarIcon = React.forwardRef(
+  (
+    {
+      size = AvatarIconSize.Md,
+      color = TextColor.primaryDefault,
+      backgroundColor = BackgroundColor.primaryMuted,
+      className = '',
+      iconProps,
+      iconName,
+      ...props
+    }: AvatarIconProps,
+    ref: Ref<HTMLElement>,
+  ) => {
+    return (
+      <AvatarBase
+        ref={ref}
+        size={size}
+        display={DISPLAY.FLEX}
+        alignItems={AlignItems.center}
+        justifyContent={JustifyContent.center}
+        color={color as TextColor}
+        backgroundColor={backgroundColor}
+        borderColor={BorderColor.transparent}
+        className={classnames('mm-avatar-icon', className)}
+        {...props}
+      >
+        <Icon
+          color={IconColor.inherit}
+          name={iconName}
+          size={size as unknown as IconSize}
```

Comment:
hey @georgewrmarshall, what do you think about this? I have done this to avoid TS error since 'size' of IconProps only accepts IconSize.

Reply:
Hmm I see. Could we create a mapping of the AvatarIconSize to the IconSize enum somehow?

```
const avatarSizeToIconSizeMap: Record<AvatarIconSize, IconSize> = {
  [AvatarIconSize.Xs]: IconSize.Xs,
  [AvatarIconSize.Sm]: IconSize.Sm,
  [AvatarIconSize.Md]: IconSize.Md,
  [AvatarIconSize.Lg]: IconSize.Lg,
  [AvatarIconSize.Xl]: IconSize.Xl,
};

const AvatarIcon: React.FC<AvatarIconProps> = ({ size }) => {
  const iconSize = avatarSizeToIconSizeMap[size]; // Map AvatarIconSize to IconSize

  return <Icon size={iconSize} />;
};
```